dashboards-observability/public/components/visualizations/charts/bar/bar.tsx

```diff
@@ -27,27 +29,19 @@ export const Bar = ({ visualizations, layout, config }: any) => {
     dataConfig.valueOptions && dataConfig.valueOptions.xaxis ? dataConfig.valueOptions.xaxis : [];
   const yaxis =
     dataConfig.valueOptions && dataConfig.valueOptions.xaxis ? dataConfig?.valueOptions.yaxis : [];
-  const barOrientation =
-    dataConfig?.chartOptions?.orientation &&
-    dataConfig.chartOptions.orientation[0] &&
-    dataConfig.chartOptions.orientation[0].orientationId
-      ? dataConfig.chartOptions.orientation[0].orientationId
-      : visualizations.vis.orientation;
+  const barOrientation = dataConfig?.chartStyles?.orientation || vis.orientation;
   const { defaultAxes } = visualizations.data;
+  const tickAngle = dataConfig?.chartStyles?.rotateBarLabels || vis.labelAngle
+  const lineWidth = dataConfig?.chartStyles?.lineWidth || vis.lineWidth;
+  const fillOpacity = dataConfig?.chartStyles?.fillOpacity !== undefined ? dataConfig?.chartStyles?.fillOpacity / FILLOPACITY_DIV_FACTOR : vis.fillOpacity / FILLOPACITY_DIV_FACTOR;
+  const barWidth = 1 - (dataConfig?.chartStyles?.barWidth || vis.barWidth);
+  const groupWidth = 1 - (dataConfig?.chartStyles?.groupWidth || vis.groupWidth);
+  const isVertical = barOrientation === vis.orientation;
+  const showLegend = dataConfig?.legend?.showLegend && dataConfig.legend.showLegend !== vis.showLegend ? false : true;
```

Comment:
This should work. We usually do not assign true or false after condition. The condition itself returns a boolean value.
```JS
const showLegend = !(dataConfig?.legend?.showLegend && dataConfig.legend.showLegend !== vis.showLegend);
```

Reply:
Yes, it's working. Thanks!